File: Accenture/4. Maximum Number/code.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;
int maxCandies(vector<int>& A, int M) {
    sort(A.begin(), A.end());
    int count = 0;
    for (int price : A) {
        if (price % 5 == 0) {
            count++;
        } else if (price <= M) {
            count++;
            M -= price;
        }}
    return count;
}
int main() {
    vector<int> A = {5, 10, 12, 15, 7, 6};
    int M = 20;
    cout << maxCandies(A, M) << endl;  // Output: 5
    return 0;
}

File: Accenture/4. Maximum Number/code.py
def max_candies(A, M):
    A.sort()
    count = 0
    for price in A:
        if price % 5 == 0:
            count += 1
        elif price <= M:
            count += 1
            M -= price
    return count
# Test case
A = [5, 10, 12, 15, 7, 6]
M = 20
print(max_candies(A, M))  # Output: 5
